app/src/main/java/is/genki/bigredapp/android/DiningLocationActivity.java

```diff
@@ -0,0 +1,184 @@
+package is.genki.bigredapp.android;
+
+import android.app.AlertDialog;
+import android.support.v7.app.ActionBarActivity;
+import android.support.v7.app.ActionBar;
+import android.support.v4.app.Fragment;
+import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+import android.os.Build;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class DiningLocationActivity extends ActionBarActivity {
+
+    public static final String KEY_DINING_HALL = "DiningLocationActivity.DINING_HALL";
+    public static final String KEY_DINING_HALL_URL = "DiningLocationActivity.DINING_HALL_URL";
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_dining_location);
+        final PlaceholderFragment fragment = new PlaceholderFragment();
+        if (savedInstanceState == null) {
+            getSupportFragmentManager().beginTransaction()
+                    .add(R.id.container, fragment)
+                    .commit();
+        }
+
+        Bundle extras = getIntent().getExtras();
+        if (extras != null) {
+            final String diningHall = extras.getString(KEY_DINING_HALL);
+            String url = extras.getString(KEY_DINING_HALL_URL);
+
+            setTitle(diningHall);
+
+            if (GetRequest.isConnected(this)) {
```

Comment:
I don't know if it matters that we wait until the activity is created to fire off the HTTP request.